openai/client.py

```diff
@@ -0,0 +1,392 @@
+import time
+import typing
+
+import openai
+
+LATEST_AZURE_API_VERSION = "2023-05-15"
+
+
+class AzureTokenAuth:
+    """Authentication using an Azure AD token.
+    """
+
+    def __repr__(self):
+        return f"AzureTokenAuth({type(self._credential)})"
+
+    def __init__(self, *, credential=None):
+        """Create a new AzureTokenAuth instance. If no credential is passed, 
+        it will use ~azure.identity.DefaultAzureCredential
+        """
+        if not credential:
+            try:
+                import azure.identity
+            except ImportError:
+                raise Exception(
```

Comment:
Should we be consistent with the error that openai raises for missing dependencies? https://github.com/openai/openai-python/blob/main/openai/datalib/common.py#L16

Reply:
I don't want to import `datalib` for this. It is a bit of a stylistic choice, but I don't really see anyone catching this specific exception (it is an environment setup issue), and thus likely fatal. The alternative is to move the datalib exception somewhere else.

Reply:
Good point that no one is going to catch this